src/+replab/+dialects/Compiled.m

```diff
@@ -0,0 +1,141 @@
+            
+            % We try to use the compiled code
+            if (~self.triedBefore) || self.isWorking
+                firstPartWorks = true;
+                if ~self.triedBefore
+                    % We try to use the compiled option
+                    % Make sure we are in the current path
+                    initialPath = pwd;
+                    try
+                        cd(self.path);
+
+                        needToCompile = true;
+                        if (exist([self.fileName, '_mex.', mexext], 'file') == 2) || (exist([self.fileName, '_mex.', mexext], 'file') == 3)
+                            % If the program was already compiled, we check that the
+                            % compilation is up to date
+                            if exist([self.fileName, '_timestamp_', mexext, '.mat'], 'file')
+                                savedData = load([self.fileName, '_timestamp_', mexext, '.mat']);
+                                fileProperties = dir([self.fileName, '_mex.', self.extension]);
+                                if isequal(savedData.timestamp, fileProperties.date)
+                                    needToCompile = false;
+                                end
+                            end
+                        end
+
+                        if needToCompile
+                            % If the program was never compiled, or the source was
+                            % modified, we try to compile it
+                            mex('-largeArrayDims', [self.fileName, '_mex.', self.extension]);
+
+                            % We save the timestamp corresponding to this
+                            % compilation
+                            fileProperties = dir([self.fileName, '_mex.', self.extension]);
+                            timestamp = fileProperties.date;
+                            save([self.fileName, '_timestamp_', mexext, '.mat'], 'timestamp')
+                        end
+                    catch
+                        firstPartWorks = false;
+                    end
+                    % return to the previous path
+                    cd(initialPath);
+                    self.triedBefore = true;
+                end
+                if firstPartWorks
+                    % The preparation worked, we try call the optimized method
+                    self.isWorking = true;
+                    try
+                        command = {self.package, '.', self.fileName, '_mex(varargin{:});'};
+                        command = cat(2, command{:});
+                        [varargout{1:self.nargout}] = eval(command);
+                    catch
+                        self.isWorking = false;
+                    end
+                else
+                    self.isWorking = false;
+                end
+            end
+
+            if ~self.isWorking
+                % Inform that the method did not succeed
+                varargout{1} = replab.DispatchNext;
```

Comment:
Yes, so here actually I started to decouple the two : previously, access to the optimized code was made through the `dispatch` framework. Now it is not the case anymore: the function `replab.graph.connectedComponents` uses an `if` statement instead of `replab.dispatch.call`.

The only reason I use `replab.DispatchNext` here is as a signification that the function failed. I didn't want to create a new symbol just for that without doing the full move away from dispatch elsewhere in the code as well. It should be straightforward to update this symbol at the moment when we remove the dispatch feature.

Reply:
That works!

I realize I like using a return value instead of an exception/error; I prefer to keep exceptions/errors for exceptional error handling (i.e. bugs, incorrect input data, ...)